### PR TITLE
Pull Request for Issue2170: Energy Tracking

### PR DESCRIPTION
**Pull request by iainworkman**
- Added controls (and getters) for all the tracking PVs to PGMBeamline
- Added views for both exit slit tracking controls to the Slit Control View
- Created an Undulator view, which for now just contains a view for undulator tracking.
- Created a Grating view, which for now just contains a view for grating tracking.

### CODE REVIEW

**Comment by iainworkman** [comment]
Hey @XuetaoLiu-CLS Could you look over this one for me?

File: source/application/PGM/PGMAppController.cpp [hunk 1]
```diff
@@ -104,6 +106,9 @@ void PGMAppController::createGeneralPanes()
 	PGMSlitControlView *slitView = new PGMSlitControlView();
     QWidget *slitWidget = mw_->buildMainWindowPane("Slits", ":/utilities-system-monitor.png", slitView);
     mw_->addPane(slitWidget, "General", "Slits", ":/utilities-system-monitor.png");
+
+	mw_->addPane(mw_->buildMainWindowPane("Mono Grating", ":/utilities-system-monitor.png", new PGMGratingView), "General", "Mono Grating", ":/utilities-system-monitor.png");
+	mw_->addPane(mw_->buildMainWindowPane("Undulator", ":/utilities-system-monitor.png", new PGMUndulatorView), "General", "Undulator", ":/utilities-system-monitor.png");
```
Comment (XuetaoLiu-CLS) [inline source/application/PGM/PGMAppController.cpp]: we have attributes for the magic strings: `generalPaneCategoryName_` and `generalPaneIcon_`

File: source/ui/PGM/PGMUndulatorView.cpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#include "PGMUndulatorView.h"
+
+#include <QVBoxLayout>
+
+#include "ui/beamline/AMExtendedControlEditor.h"
+#include "beamline/PGM/PGMBeamline.h"
+
+PGMUndulatorView::PGMUndulatorView(QWidget *parent) : QWidget(parent)
```
Comment (XuetaoLiu-CLS) [inline source/ui/PGM/PGMUndulatorView.cpp]: at this moment, this class is very alike `PGMGratingView`. Maybe we should think to generalize them?

**Comment by XuetaoLiu-CLS** [comment]
Looks good other than a question on the two aliked functions

Time Estimate: 5 minutes

File: source/ui/PGM/PGMUndulatorView.cpp (hunk 2 above)
Comment (iainworkman) [inline source/ui/PGM/PGMUndulatorView.cpp]: It won't be as much once the grating and undulator are both fully implemented.

**Pull request merged**